Add WanSync docs [CN-369] (#152)
hazelcast/hazelcast-platform-operator-docs: docs/modules/ROOT/pages/wan-sync.adoc

= Syncronizing WAN replicated data

> :description: pass:q[WAN Syncronization allows you to overcome this out-of-sync issues with the WAN replicated data. ]

Reviewer: ```suggestion
:description: pass:q[WAN Syncronization allows you to overcome out-of-sync issues with the WAN replicated data. ]
```

:page-enterprise: true

{description}

See xref:hazelcast:wan:advanced-features#synchronizing-wan-clusters.adoc[Synchronizing WAN Clusters] for more detailed information about the configuration and behavior of WAN replication in Hazelcast.

== Prerequisites for WAN Sync

> Before you create and configure a `WanSync` resource, you need to create the following:

Reviewer: ```suggestion
Before you create and configure a `WanSync` resource, you must create the following:
```

- At least one `Hazelcast` resource and one `Map` resource for the source Hazelcast cluster.
- A target Hazelcast cluster that is accessible from the source cluster.      

== Configuring the WanSync Resource

> To configure WAN Syncronization, you need to create `WanSync` resource. For more detailed information, see the xref:api-ref.adoc#wansyncspec[API Reference].

Reviewer: ```suggestion
To configure WAN Syncronization, you must create `WanSync` resource. For more detailed information, see the xref:api-ref.adoc#wansyncspec[API Reference].
```

> There are two ways to create a `WanSync` resource. The first one is by spcifying the existing `WanReplication` name. This way, the `WanSync` will reuse the existing and running WAN replication proccess.

Reviewer: ```suggestion
There are two ways to create a `WanSync` resource. The first one is by specifying the existing `WanReplication` name. This means that the `WanSync` reuses the existing and running WAN replication process.
```

.Example of WanSync using the existing WanReplication
[source,yaml,subs="attributes+"]
----
include::ROOT:example$/wan-sync-from-replication.yaml[]
----

> NOTE: For using this appoach, you must make sure that the `WanReplication` resource is succesully created.

Reviewer: ```suggestion
NOTE: For this approach, you must make sure that the `WanReplication` resource is successfully created.
```

> Another apporach is to specify the WAN replication configuration directly in the `WanSync` resource. This way, the operator will create a new `WanReplication` resource before running the WAN syncronization procces. The created `WanReplication` will be deleter when the correspondig `WanSync` is deleted.

Reviewer: ```suggestion
Another approach is to specify the WAN replication configuration directly in the `WanSync` resource. This means that the operator creates a new `WanReplication` resource before running the WAN syncronization process. The created `WanReplication` is deleted when the corresponding `WanSync` is deleted.
```

.Example of WanSync with the new WanReplication configuration
[source,yaml,subs="attributes+"]
----
include::ROOT:example$/wan-sync-config.yaml[]
----

== Checking the Status of a WAN Sync

> After creating the `WanSync` resource, run the following command to find out the WAN syncronization status:

Reviewer: ```suggestion
After creating the `WanSync` resource, run the following command to see the WAN syncronization status:
```

[source,shell]
----
kubectl get wansync wansync-sample
----

> You should see the following output:

Reviewer: ```suggestion
The output is similar to the following:
```
